### PR TITLE
few tweaks to driver finding logic

### PR DESCRIPTION
**Pull request by titusfortner**
### Description
1. Don't execute Java `ExecutableFinder()` if a System Property has been defined
2. Python log results even if error code isn't 0

### Motivation and Context
Ensured logic the same between Java/Python/Ruby/.NET

### CODE REVIEW

**Review by diemol — COMMENTED** [review]
Does this mean we are letting Selenium Manager find the driver on the PATH in Java?

**Comment by titusfortner** [comment]
No, not yet. The default parameter for getProperty always evaluates, even when there is a property, which is less performant and unnecessary. So very minor performance improvement; and now we just remove those 3 lines when we do let SM get from PATH.

File: java/src/org/openqa/selenium/remote/service/DriverFinder.java [hunk 1]
```diff
@@ -14,11 +14,14 @@ public class DriverFinder {
 
   public static String getPath(DriverService service, Capabilities options) {
     Require.nonNull("Browser options", options);
-    String defaultPath = new ExecutableFinder().find(service.getDriverName());
```
Comment (diemol) [inline java/src/org/openqa/selenium/remote/service/DriverFinder.java:17]: This line is the one that finds the driver on the PATH, and it is getting removed. Right?

Comment (titusfortner) [reply java/src/org/openqa/selenium/remote/service/DriverFinder.java:17]: no, it moved to line 20. I moved it after `getProperty()` with a conditional to only execute if there was no property found.

Comment (diemol) [reply java/src/org/openqa/selenium/remote/service/DriverFinder.java:17]: You are right.

**Review by diemol — APPROVED** [review]

**Pull request merged**